Core.ApplicationServices/Model/Contracts/ContractPermissions.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Core.ApplicationServices.Authorization;
+
+namespace Core.ApplicationServices.Model.Contracts
+{
+    public class ContractPermissions
+    {
+        public ResourcePermissionsResult BasePermissions { get; }
+
+        public ContractPermissions(ResourcePermissionsResult basePermissions)
+        {
+            BasePermissions = basePermissions;
```

Comment:
Don't we have any special access rights/permissions besides "anyone in the same org can see all contracts"?
I'm wondering why there is no existing model for this?
Has it been spread out in the domain or elsewhere? Eg ItContractRight (this is wrt roles I know, but just as an example)

Reply:
SystemPermissions also contain e.g. DeleteConflicts.
The permissions are being calculated based on the role of the current user, so the result is going to be different for LocalAdmin and User.

The base permission type is the ResourcePermissionsResult (which is also used in SystemUsagePermissions and SystemPermissions), which is then wrapped in the {Module}Permissions class. This allows us to then further expand the permissions we return in case of a specific use case, like with the delete conflicts, where we show the reason why user cannot delete the system

Reply:
Talk on anemic datamodel - to be discussed later (perhaps)